### PR TITLE
Use `attrs` for marker constructor

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Use `attrs` for the marker constructor, to simplify things, and ensure the code is simple enough. 

I did this whilst trying to solve #49. Unfortunately `attrs` won't solve the issue, but it makes `Marker` look a bit nicer!

Also had to rework the way the cache around pose estimation worked, as the `attrib` objects aren't hashable, meaning `Marker` isn't hashable, meaning it can't be used in the cache. The new way is a bit nicer, anyway.

### CODE REVIEW

File: zoloto/marker.py [hunk 1]
```diff
@@ -58,13 +54,12 @@ def spherical(self):
         x, y, z = self._tvec
         return Spherical(rot_x=arctan2(y, z), rot_y=arctan2(x, z), dist=self.distance)
 
-    @property
+    @cached_property
     def cartesian(self):
         return ThreeDCoordinates(*self._tvec)
 
-    @clru_cache(maxsize=None)
```
Comment (trickeydan) [inline zoloto/marker.py]: Why aren't we caching the pose vectors any more?

Comment (RealOrangeOne) [reply zoloto/marker.py]: We are, manually. Notice the assignment at the bottom of the method. This now links the cache to the marker instance, and so is de-allocated when the marker drops out of scope

**Comment by RealOrangeOne** [comment]
> this is a breaking change

It is, but upon reflection there's ways I can do this without breaking the API, so i'll do that before this is actually ready.

**Comment by trickeydan** [comment]
I know this'll need rebasing, but having looked at the typing stuff, this will make typing much easier.

I think it's probably best to get this merged before I delve into adding lots of type hints.

**Comment by RealOrangeOne** [comment]
I'm planning to do a large refactor of the `Marker` class to make it infinitely easier to comprehend and work with, because currently even I find it a bit weird.

With that, it _might_ migrate to `attrs` it might not. This PR shows the code cleanup is minimal, so it mostly comes down to if it actually looks any nicer.

**Pull request closed without merging**